cache/ttlmemory.go

```diff
@@ -0,0 +1,57 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NewTTLInMemory creates a new thread-safe in-memory ttl cache.
```

Comment:
with OTel tracing?

Reply:
I did the other suggestion of removing otel tracing by default and making it an exported wrapper call